File: leetcode/2834.找出美丽数组的最小和.py
class Solution:
    def minimumPossibleSum(self, n: int, target: int) -> int:
        vis, ans, p, cnt = set(), 0, 1, 0
        while cnt < n:
            if p not in vis:
                ans += p
                if target - p > 0:
                    vis.add(target - p)
                p+=1
                cnt += 1
            else:
                p+=1
        return ans

File: leetcode/2834.找出美丽数组的最小和.go
func minimumPossibleSum(n int, target int) int {
    MOD := 1000_000_007
    k := target / 2
    if k > n {
        k = n
    }
    d := n - k
    return ((1 + k) * k / 2 + max(0, (target + target + d - 1) * d / 2)) % MOD
}